src/main/java/com/training/web/pages/flipkart/ResultPage.java

```diff
@@ -1,93 +1,92 @@
 package com.training.web.pages.flipkart;
 
+import com.codeborne.selenide.SelenideElement;
 import com.training.basepages.FlipkartBasePage;
-import org.openqa.selenium.By;
-import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.WebElement;
-import org.openqa.selenium.support.ui.ExpectedConditions;
-import org.openqa.selenium.support.ui.WebDriverWait;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import java.text.NumberFormat;
+import java.text.ParseException;
 import java.util.*;
+import static com.codeborne.selenide.Selenide.*;
+import static com.codeborne.selenide.Condition.*;
+import static com.codeborne.selenide.WebDriverRunner.getWebDriver;
 
 public class ResultPage extends FlipkartBasePage {
     private final Logger logger = LoggerFactory.getLogger(this.getClass());
 
-    private By lowToHigh = By.xpath("//*[text()='Price -- Low to High']");
-    private By productResults = By.xpath("//img[contains(@class,'_2r_T1I')]");
-    private By shoesPrice = By.className("_30jeq3");
-    private By nextPageButton = By.xpath("//*[text()='Next']");
-    private By products = By.xpath("//div[@class='_312yBx SFzpgZ']");
-
-    public ResultPage (WebDriver driver) {
-        super(driver);
-    }
+    private String lowToHigh = "//*[text()='Price -- Low to High']";
+    private String productResults = "//img[contains(@class,'_2r_T1I')]";
+    private String shoesPrice = "//*[@class='_30jeq3']";
+    private String nextPageButton = "//*[text()='Next']";
+    private String sortBy = "//*[@class='_10Ermr']";
 
     public ResultPage sortLowToHigh() {
-        wait.until(ExpectedConditions.elementToBeClickable(lowToHigh));
-        driver.findElement(lowToHigh).click();
-        wait.until(ExpectedConditions.elementToBeClickable(shoesPrice));
+
+        waitForLoader();
+        $x(lowToHigh).shouldBe(visible).click();
         return this;
     }
 
-    public ArrayList<Integer> getPrice() {
-        waitForLoader();
-        wait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(shoesPrice));
+    public List<SelenideElement> getPriceString(){
+        $x(sortBy).shouldBe(visible);
+        List<SelenideElement> list= $$x(shoesPrice);
+        return list;
+    }
 
-        List<WebElement> list = driver.findElements(shoesPrice);
-        logger.info("Size of list " + (list.size()));
+    public ArrayList<Integer> getPrice() throws ParseException {
 
-        ArrayList<Integer> priceList = new ArrayList<>();
-        for (WebElement listElement : list) {
-            priceList.add(Integer.parseInt(listElement.getText().substring(1)));
+        $x(nextPageButton).shouldBe(visible);
+        ArrayList<Integer> priceList= new ArrayList<>();
+        logger.info("Size of list"+getPriceString().size());
+        for (SelenideElement listItem:getPriceString()){
```

Comment:
I don't think the purpose of the logger is being solved here - because `getPriceString()` is going to collect the list for each of its call.
So every-time you call `getPriceString()` - it is going trigger finding of elements.